apps/shell/tash_security.c

```diff
@@ -23,43 +23,129 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <mbedtls/sha256.h>
+#include <mbedtls/pkcs5.h>
+#include <security/security_crypto.h>
 #include "tash_internal.h"
 
-#define SHA256_HASH_nCHAR  (64) // 32 bytes
-#define SHA256_HASH_nBYTE  (32) // 32 bytes
-#define CONVERT_nCHAR      (sizeof(unsigned long) * 2) // unsigned long for strtoul
+#define CONVERT_nCHAR			(sizeof(unsigned long) * 2) // unsigned long for strtoul
+
+#define PBKDF2_SALT_ENCRYPTED	"\x50\x3B\x3C\x3E\xBF\x0D\x3D\x2C\x9D\x6E\xA2\xF3\x62\xF5\x0F\x9F"
+#define PBKDF2_SALT_LEN			(16)
+#define PBKDF2_KEY_nCHAR		(64) // 32 bytes
+#define PBKDF2_KEY_LEN			(32) // 32 bytes
+#define PBKDF2_ITERATION		(777)
+
+#define AES_BLOCK_SIZE			(16)
+#define AES128_KEY				"ss/1"
+
+
+security_error salt_aes_decrypt(char *input, unsigned int input_size, char *output, unsigned int output_size) 
```

Comment:
I think there's no reason for security_error type because it defined in lower layer.
tash should have it's own type instead of lower's